### DISCUSSION TITLE
Copilot chat not working, however normal copilot is fine. (VScode | network proxy )

### DISCUSSION THREAD

**Discussion by JiveyGuy** [community/community#109362] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

Using VScode behind network proxy, copilot works well with win-ca extension. However using copilot chat is a a fail with Copilot responding: token expired or invalid: 403. 

Followed advice of signing out and reloading then back in, no change. 

Disabled strict SSL cert, no change. 

Any advice? Can I inject a self signed cert I have downloaded into copilot?

**Comment by JiveyGuy** [comment]
Also getting "failed to make request" for **Quick Fix...**

**Comment by davecheney** [comment]
@JiveyGuy your version of chat is very old. Please

1. Switch from Insiders (no longer required for chat) to regular VSCode
2. Install the copilot extension from the marketplace, which will automatically install the chat sub extension.

**Reply by JiveyGuy** [reply]
Hi @davecheney thanks so much for taking the time to help out. 

I have many VScode installs, and one insiders install, here is the about info for the one that has this issue: 

```cfg
Version: 1.86.2 (system setup)
Commit: 903b1e9d8990623e3d7da1df3d33db3e42d80eda
Date: 2024-02-13T19:40:56.878Z
Electron: 27.2.3
ElectronBuildId: 26908389
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Windows_NT x64 10.0.19045
```

I installed from the market place and have these diagnostics from Copilot:

```MD
## Copilot

- Version: 1.162.0
- Build: prod
- Editor: vscode/1.86.2

## Environment

- http_proxy: n/a
- https_proxy: n/a
- no_proxy: n/a
- SSL_CERT_FILE: n/a
- SSL_CERT_DIR: c:\Users\<redacted>\.vscode\extensions\ukoloff.win-ca-3.5.1\node_modules\win-ca\pem
- OPENSSL_CONF: n/a

## Feature Flags

- Custom Certificates: disabled
- Send Restricted Telemetry: disabled
- Chat: enabled

## Node setup

- Number of root certificates: 137
- Operating system: Windows_NT
- Operating system version: 10.0.19045
- Operating system architecture: x64
- NODE_OPTIONS: n/a
- NODE_EXTRA_CA_CERTS: n/a
- NODE_TLS_REJECT_UNAUTHORIZED: n/a
- tls default min version: TLSv1.2
- tls default max version: TLSv1.3

## Network Configuration

- Proxy host: n/a
- Proxy port: n/a
- Kerberos SPN: n/a
- Reject unauthorized: disabled
- Fetcher: HelixFetcher

## Reachability

- github.com: HTTP 200 - OK
- copilot-proxy.githubusercontent.com: HTTP 200 - OK
- api.githubcopilot.com: HTTP 403 - Forbidden
- default.exp-tas.com: HTTP 200 - OK

## VS Code Configuration

- HTTP proxy: 
- HTTP proxy authentication: n/a
- Proxy Strict SSL: true
- Extension HTTP proxy support: override

## Extensions

- Is `win-ca` installed?: true
- Is `mac-ca` installed?: false

## Authentication

- GitHub username: JiveyGuy
```

I see that `api.githubcopilot.com: HTTP 403 - Forbidden` is showing. Is there a way to overcome this?

**Reply by davecheney** [reply]
Is it possible that your employer, internet service provider, or government, may be intercepting your SSL traffic?

**Reply by JiveyGuy** [reply]
That is very possible, we are behind a proxy. I have a self signed cert .pem I was wondering how to use that for chat because normal copilot is using it no problem. 

Is there a way to route SSL traffic to a different address the redirect it to GitHub?

**Reply by davecheney** [reply, marked as answer]
I'm afraid not, api.githubcopilot.com is required for chat.
